internal/pkg/coordinator/monitor.go

```diff
@@ -512,6 +512,12 @@ func runUnenrollerWork(ctx context.Context, bulker bulk.Bulk, policyID string, u
 		return err
 	}
 
+	if len(agents) == 0 {
+		zlog.Info().
+			Msg("no agents to unenroll")
+		return nil
+	}
+
```

Comment:
[Blocker]
See comment below

```suggestion
		if errors.Is(err, dl.ErrNotFound) {
			zlog.Info().Msg("no agents to unenroll")
			return nil
		}

		return err
	}
```